pastas/stressmodels.py

```diff
@@ -1151,3 +1153,170 @@ def to_dict(self, series=True):
+    functions. When the simulation reaches the second drainage level, the
+    second response function becomes active. Because ofits structure,
+    TarsoModel cannot be combined with other stressmodels, a constant or a
+    transform.
+    TarsoModel inherits from RechargeModel. Only parameters specific to the
+    child class are named above.
+    
+    References
+    ----------
+    .. [1] Knotters, M. & De Gooijer, Jan G.. (1999). TARSO modeling of water table depths. Water Resources Research. 35. 10.1029/1998WR900049. 
+
+    """
+    _name = "TarsoModel"
+    def __init__(self, prec, evap, oseries=None, dmin=None, dmax=None,
+                 rfunc=Exponential, **kwargs):
+        if oseries is not None:
+            if dmin is not None or dmax is not None:
+                msg = 'Please specify either oseries or dmin and dmax'
+                raise(Exception(msg))
+            o = TimeSeries(oseries).series
+            dmin = o.min()
+            dmax = o.max()
+        elif dmin is None or dmax is None:
+            msg = 'Please specify either oseries or dmin and dmax'
+            raise(Exception(msg))
+        if not issubclass(rfunc, Exponential):
+            raise NotImplementedError("TarsoModel only supports rfunc "
+                                      "Exponential!")
+        self.dmin = dmin
+        self.dmax = dmax
+        super().__init__(prec, evap, rfunc=rfunc, **kwargs)
+
+    def set_init_parameters(self):
+        # parameters for the first drainage level
+        p0 = self.rfunc.get_init_parameters(self.name)
+        one = One(meanstress = self.dmin + 0.5 * (self.dmax - self.dmin))
+        pd0 = one.get_init_parameters(self.name).squeeze()
+        p0.loc['{}_d'.format(self.name)] = pd0
+        p0.index = ['{}0'.format(x) for x in p0.index]
+        
+        # parameters for the second drainage level
+        p1 = self.rfunc.get_init_parameters(self.name)
+        initial = self.dmin + 0.75 * (self.dmax - self.dmin)
+        pd1 = Series({'initial':initial, 'pmin':self.dmin, 'pmax':self.dmax,
+                      'vary':True, 'name':self.name})
+        p1.loc['{}_d'.format(self.name)] = pd1
+        p1.index = ['{}1'.format(x) for x in p1.index]
+        
+        # parameters for the recharge-method
+        pr = self.recharge.get_init_parameters(self.name)
+        
+        # combine all parameters
+        self.parameters = concat([p0, p1, pr])
+
+    def simulate(self, p=None, tmin=None, tmax=None, freq=None, dt=1):
+        stress = self.get_stress(p=p, tmin=tmin, tmax=tmax, freq=freq)
+        h = self.tarso(p[:-self.recharge.nparam], stress.values, dt)
+        sim = Series(h, name=self.name, index=stress.index)
+        return sim
+
+    def to_dict(self, series=True):
+        data = super().to_dict(series)
+        data['dmin'] = self.dmin
+        data['dmax'] = self.dmax
+        return data
+
+    def check_stressmodel_compatibility(self, ml):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/pastas/pastas/pullRequest?prid=5245245)